shared/src/simulation/activities/reassign-treatments.ts

```diff
@@ -0,0 +1,295 @@
+import { IsUUID } from 'class-validator';
+import type { Material, Personnel } from '../../models';
+import { Patient, SimulatedRegion } from '../../models';
+import type { PatientStatus, PersonnelType } from '../../models/utils';
+import { getCreate } from '../../models/utils';
+import type { ExerciseState } from '../../state';
+import type { CatersFor } from '../../store/action-reducers/utils/calculate-treatments';
+import {
+    removeTreatmentsOfElement,
+    tryToCaterFor,
+} from '../../store/action-reducers/utils/calculate-treatments';
+import type { Mutable } from '../../utils';
+import { UUID, uuidValidationOptions } from '../../utils';
+import { IsValue } from '../../utils/validators';
+import type { TreatPatientsBehaviorState } from '../behaviors';
+import { TreatmentProgressChangedEvent } from '../events';
+import { sendSimulationEvent } from '../utils';
+import type {
+    SimulationActivity,
+    SimulationActivityState,
+} from './simulation-activity';
+
+export class ReassignTreatmentsActivityState
+    implements SimulationActivityState
+{
+    @IsValue('reassignTreatmentsActivity' as const)
+    public readonly type = 'reassignTreatmentsActivity';
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly id!: UUID;
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly behaviorId!: UUID;
```

Comment:
```suggestion
    @IsUUID(4, uuidValidationOptions)
    public readonly id: UUID;

    @IsUUID(4, uuidValidationOptions)
    public readonly behaviorId: UUID;
```

I believe you don't need these, as they are assigned in the constructor.

If you use the Non-null assertion operator here, you don't get a type error if you forget to assign it in the constructor.

Similar for all the other such classes.